File: 1.5-ArrayBasics/ArrayBasics_in_CPP/2-ReverseTheArray.cpp
#include<bits/stdc++.h>
using namespace std;

void printArray(int arr[], int n){
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

void swap(int& a, int&b){
    int temp = a;
    a = b;
    b = temp;
}

void reverseArray(int arr[], int n){
    int start = 0;
    int end = n - 1;

    while(start <= end){
        swap(arr[start], arr[end]);
        start++;
        end--;
    }
}

int main() {

    int arr[5] = {5,4,3,2,1};
    int brr[6] = {16,15,14,13,12,11};

    reverseArray(arr,5);

    reverseArray(brr,6);

    printArray(arr,5);
    printArray(brr,6);

}

File: 1.5-ArrayBasics/ArrayBasics_in_Python/2-ReverseTheArray.py
def printArray(arr):
    for i in range(len(arr)):
        print(arr[i], end=" ")
    print('\n')

def reverseArray(arr):
    start = 0
    end = len(arr) - 1

    while(start <= end):
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1


arr = [10,9,8,7,6]

brr = [55,54,53,52,51,50]

reverseArray(arr)

reverseArray(brr)

printArray(arr)

printArray(brr)
